lib/rouge/lexers/yara.rb

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class YARA < RegexLexer
+      title "YARA"
+      desc ""
+      tag 'yara'
+      filenames '*.yar', '*.yara'
+
+      def self.keywords
+        @keywords ||= Set.new %w(
+          all and any ascii at condition contains entrypoint
+          filesize for fullword global import in include matches
+          meta not nocase of or private rule strings them wide xor
+        )
+      end
+
+      state :root do
+        rule /[^\S]+/, Text
+        rule /\/\/.*$/, Comment::Single
+        rule /\/\*/, Comment::Multiline, :block_comment
+
+        rule /(\$[a-zA-Z0-9_]+)(\s*)(=)(\s*)({)/ do
+          groups Name::Variable, Text, Text, Text, Str
```

Comment:
Tokenising whitespace as `Text` is fine but the `=` should be `Punctuation`. For the opening `{`, I'd suggest tokenising that as `Str::Other`.